src/notebooks/types.ts

```diff
@@ -26,14 +26,14 @@ export interface INotebookControllerManager {
         controller: IVSCodeNotebookController;
         selected: boolean;
     }>;
+    readonly onNotebookControllersLoaded: Event<Readonly<IVSCodeNotebookController[]>>;
```

Comment:
I debated back and forth a bit on this. I like keeping the commands independent from the notebook controller (at a basic level they could just be a small sub class or even functional) so they have their own service. However to keep the logic for when to show they need to know what controllers have been loaded. Previously we were not looking to expose this, however we're already using 'getRegisteredNotebookControllers' in product code, and it makes sense to me that other services could see a readonly list of what controllers are registered.

Reply:
👏 how notebook controllers are registered (through detection or an async command/resolver) is irrelevant to externals IMHO so exposing events for `notebookControllers` make better sense than exposing another service/command.